### PR TITLE
Deprecation warning for default Instrument method

### PR DESCRIPTION
**Pull request by aburrell**
# Description

Added a deprecation warning for Instruments that have a `default` method.  Addresses #684.

## Type of change

Please delete options that are not relevant.

- This change requires a documentation update

# How Has This Been Tested?
Added a unit test

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [N/A] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by aburrell** [comment]
Ok, for some reason this is failing in python 2.7 on Travis.  It passes locally on 2.7 and 3.7 so... not sure how to proceed.

**Review by rstoneback — APPROVED** [review]
Thanks @aburrell

**Comment by aburrell** [comment]
Ummm.... @rstoneback not sure you should approve this, given my concern above.

**Comment by rstoneback** [comment]
> Ummm.... @rstoneback not sure you should approve this, given my concern above.

That makes sense. I'm not sure how I missed all of that.

**Comment by rstoneback** [comment]
> Ok, for some reason this is failing in python 2.7 on Travis. It passes locally on 2.7 and 3.7 so... not sure how to proceed.

Best guess on my end is that the warnings module is filtering out the DeprecationWarning. DeprecationWarnings are hidden by default except for some situations, https://www.python.org/dev/peps/pep-0565/#new-default-warnings-filter-entry 
It doesn't seem to like the one in `assign_funcs`. If I do a
```
            wstr = ''.join(("The Instrument method `default` has been ",
                            "renamed `preprocess` in pysat 3.0.0. ",
                            "If this is not a pysat-managed Instrument,",
                            " you will need to update this when ",
                            "migrating to pysat 3.0.0."))
            warnings.simplefilter('always', DeprecationWarning)
            warnings.warn(wstr, DeprecationWarning, stacklevel=2)
```
then it does work.

File: pysat/_instrument.py [hunk 1]
```diff
@@ -862,6 +862,13 @@ def _assign_funcs(self, by_name=False, inst_module=None):
                                             '{:s}every instrument.'.format(estr))))
         try:
             self._default_rtn = inst.default
+            
+            warnings.warn(''.join(["The Instrument method `default` has been ",
+                                   "renamed `preprocess` in pysat 3.0.0. ",
+                                   "If this is not a pysat-managed Instrument,",
+                                   " you will need to update this when ",
+                                   "migrating to pysat 3.0.0."]),
+                          DeprecationWarning, stacklevel=2)
```
Comment (aburrell) [inline pysat/_instrument.py:871]:
```suggestion
            wstr = ''.join(("The Instrument method `default` has been ",
                            "renamed `preprocess` in pysat 3.0.0. ",
                            "If this is not a pysat-managed Instrument,",
                            " you will need to update this when ",
                            "migrating to pysat 3.0.0."))
            warnings.simplefilter('always', DeprecationWarning)
            warnings.warn(wstr, DeprecationWarning, stacklevel=2)
```

**Pull request merged**